server/src-lib/Hasura/GraphQL/Schema/Remote.hs

```diff
@@ -213,20 +270,71 @@ remoteSchemaInterface schemaDoc defn@(G.InterfaceTypeDefinition description name
           Just _  -> throw400 RemoteSchemaError $ "Interface type " <> squote name <>
             " can only include object types. It cannot include " <> squote objectName
 
+    -- 'constructInterfaceQuery' constructs a remote interface query.
+    constructInterfaceSelectionSet
+      :: [(G.Name, [Field NoFragments G.Name])]
+      -> SelectionSet NoFragments G.Name
+    constructInterfaceSelectionSet objNameAndFields =
+      let -- common interface fields that exist in every
+          -- selection set provided
+          -- #1 of Note [Querying remote schema Interfaces]
+          commonInterfaceFields =
+            Map.elems $
+            Map.mapMaybe allTheSame $
+            Map.groupOn G._fName $
+            concatMap (\(_, flds) ->
+                         filter ((`elem` interfaceFieldNames) . G._fName) flds)
+                      $ objNameAndFields
+
+          interfaceFieldNames = map G._fldName fields
+
+          allTheSame (x:xs) | all (== x) xs = Just x
+          allTheSame _ = Nothing
+
+          -- #2 of Note [Querying remote schema interface fields]
+          nonCommonInterfaceFields =
+            catMaybes $ flip map objNameAndFields $ \(objName, flds) ->
+              let nonCommonFields = filter (not . flip elem commonInterfaceFields) flds
```

Comment:
```suggestion
            catMaybes $ flip map objNameAndFields $ \(objName, fields) ->
              let nonCommonFields = filter (not . flip elem commonInterfaceFields) fields
```

Reply:
I was about to use `fields` too but it clashes with the `fields` defined in the arguments defined in 

```haskell

remoteSchemaInterface
  :: forall n m
  . (MonadSchema n m, MonadError QErr m)
  => SchemaIntrospection
  -> G.InterfaceTypeDefinition [G.Name]
  -> m (Parser 'Output n (G.SelectionSet NoFragments G.Name))
remoteSchemaInterface schemaDoc defn@(G.InterfaceTypeDefinition description name _directives fields possibleTypes) =
```

Reply:
Changing `flds` to `objFields`

Reply:
Ah, I see :)